test/exoscale/vinyl/demostore.clj

```diff
@@ -44,7 +51,11 @@
                             :on [:group-by "path" "bucket"]
                             :type :count-not-null}
                            {:name "bucket_paths"
-                            :on [:concat "bucket" "path"]}]}})
+                            :on [:concat "bucket" "path"]}]}
+   :City   {:primary-key [:concat :type-key [:nest "location" "name"]
+                                            [:nest "location" "zip_code"]]}})
```

Comment:
This one can be changed but it will be a breaking change.
I changed the querying capability, not the way schema can be defined. This supported `nest` already.

Reply:
Given the fact that there is a single consuming schema for now I think this is safe.